From 440e54eb676a63ce4e32f85f3ff2ea0f9aed0d1c Mon Sep 17 00:00:00 2001
From: Cyril Hrubis <chrubis@suse.cz>
Date: Tue, 17 Sep 2013 15:55:53 +0200
Subject: [PATCH] syscalls: syslog: Fix daemon restart on SUSE.

Some distributions (at least SUSE) names the service 'syslog' even if
the syslog binary is named syslog-ng or rsyslog.

This patch adds codepath that tries restart syslog service if anything
else has failed.

Signed-off-by: Cyril Hrubis <chrubis@suse.cz>
---
 testcases/kernel/syscalls/syslog/syslog-lib.sh | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/testcases/kernel/syscalls/syslog/syslog-lib.sh b/testcases/kernel/syscalls/syslog/syslog-lib.sh
index 0885805..9c5d7f1 100755
--- a/testcases/kernel/syscalls/syslog/syslog-lib.sh
+++ b/testcases/kernel/syscalls/syslog/syslog-lib.sh
@@ -124,7 +124,16 @@ restart_syslog_daemon()
 			# then it needs to be fixed.
 			sleep 2
 		else
-			$cleanup_command
+			#
+			# Some distributions name the service syslog even if
+			# the package is syslog-ng or rsyslog, so try it once
+			# more with just syslog.
+			#
+			restart_daemon "syslog"
+
+			if [ $? -ne 0 ]; then
+				$cleanup_command
+			fi
 		fi
 	fi
 }
-- 
1.8.1.5

